packages/web3-eth-contract/src/contract.ts

```diff
@@ -955,37 +973,73 @@ export class Contract<Abi extends ContractAbi>
 		this._jsonInterface = [...result] as unknown as ContractAbiWithSignature;
 	}
 
+	// eslint-disable-next-line class-methods-use-this
+	private _getAbiParams(abi: AbiFunctionFragment, params: unknown[]): Array<unknown> {
+		try {
+			return validatorUtils.transformJsonDataToAbiFormat(abi.inputs ?? [], params);
+		} catch (error) {
+			throw new Web3ContractError(
+				`Invalid parameters for method ${abi.name}: ${(error as Error).message}`,
+			);
+		}
+	}
 	private _createContractMethod<T extends AbiFunctionFragment, E extends AbiErrorFragment>(
 		abi: T,
 		errorsAbis: E[],
 	): ContractBoundMethod<T> {
 		return (...params: unknown[]) => {
 			let abiParams!: Array<unknown>;
+			const abis = this._overloadedMethodAbis.get(abi.name) ?? [];
+			let methodAbi: AbiFunctionFragment = abis[0];
 
-			try {
-				abiParams = validatorUtils.transformJsonDataToAbiFormat(abi.inputs ?? [], params);
-			} catch (error) {
-				throw new Web3ContractError(
-					`Invalid parameters for method ${abi.name}: ${(error as Error).message}`,
-				);
-			}
+			const arrayOfAbis: AbiFunctionFragment[] = abis.filter(
+				_abi => (_abi.inputs ?? []).length === params.length,
+			);
 
-			validator.validate(abi.inputs ?? [], abiParams);
+			if (abis.length === 1 || arrayOfAbis.length === 0) {
+				abiParams = this._getAbiParams(methodAbi, params);
+				validator.validate(abi.inputs ?? [], abiParams);
+			} else {
+				const errors: Web3ValidationErrorObject[] = [];
+
+				for (const _abi of arrayOfAbis) {
+					try {
+						abiParams = this._getAbiParams(_abi, params);
+						validator.validate(
+							_abi.inputs as unknown as ValidationSchemaInput,
+							abiParams,
+						);
+						methodAbi = _abi;
+						break;
```

Comment:
why iterations are breaking here?

Reply:
If any of the validation is successful, we do not need to check other